File: scala-blueprints/src/main/scala/idesyde/blueprints/ExplorationRequestMessage.scala
package idesyde.blueprints

import upickle.default._
import idesyde.core.Explorer
import idesyde.core.ExplorerConfiguration

final case class ExplorationRequestMessage(
    model_message: DecisionModelMessage,
    previous_solutions: Set[ExplorationSolutionMessage],
    configuration: ExplorerConfiguration
) derives ReadWriter

object ExplorationRequestMessage {
  def fromJsonString(s: String): ExplorationRequestMessage = read(s)
}

File: java-blueprints/src/main/java/idesyde/blueprints/ExplorationRequestMessage.java
package idesyde.blueprints;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import idesyde.core.ExplorationSolution;
import idesyde.core.Explorer;

import java.util.Set;

@JsonSerialize
public record ExplorationRequestMessage(
                @JsonProperty("model_message") DecisionModelMessage modelMessage,
                @JsonProperty("previous_solutions") Set<ExplorationSolution> previousSolutions,
                Explorer.Configuration configuration) {
}
